source/initial_composition/random_perturbation.cc

```diff
@@ -0,0 +1,95 @@
+/*
+  Copyright (C) 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/initial_composition/random_perturbation.h>
+
+namespace aspect
+{
+  namespace InitialComposition
+  {
+    template <int dim>
+    RandomPerturbation<dim>::
+    RandomPerturbation ()
+      :
+      random_number_generator(5432)
+    {
+    }
+
+    template <int dim>
+    double
+    RandomPerturbation<dim>::
+    initial_composition (const Point<dim> &/*position*/,
+                         const unsigned int /*compositional_index*/) const
+    {
+      // Uniform distribution on the interval [0,1]. This
+      // will be used to generate random particle locations.
+      boost::uniform_01<double> uniform_distribution_01;
+
+      return magnitude * 2.0 * (uniform_distribution_01(random_number_generator) - 0.5);
```

Comment:
I will admit that I do not know much about the thread-safety of boost random functions. https://stackoverflow.com/questions/2920497/thread-safty-of-boost-rng seems to suggest they are not thread-safe, and https://stackoverflow.com/questions/21237905/how-do-i-generate-thread-safe-uniform-random-numbers has suggestions on how to make a thread safe function, but it requires macros for different compilers (and I do not know how this would interact with less common compilers like intel or cray). Do you have a suggestion for a thread-safe version? Does it matter? I am not too concerned about the quality of the randomness, I just want some non-regular perturbations.